src/frontend/screens/Settings/MapSettings/MapImportBottomSheet.tsx

```diff
@@ -0,0 +1,322 @@
+const BackdropNonPressable = (props: BottomSheetBackdropProps) => {
+  return (
+    <BottomSheetBackdrop {...props} opacity={0.3} pressBehavior={"none"} />
+  );
+};
+
+const MapImportBottomSheet = React.forwardRef<MapImportBottomSheetMethods, {}>(
+  (_props, ref) => {
+    const [state, setState] = React.useState<BottomSheetStates>("select");
+    const bottomSheetRef = React.useRef<BottomSheetMethods>(null);
+    const { snapPoints, updateSheetHeight } = useSnapPointsCalculator();
+    const { add: addMapImport } = useMapImportsManager();
+
+    React.useImperativeHandle(ref, () => {
+      return {
+        open: () => bottomSheetRef.current?.expand(),
+      };
+    });
+
+    const closeSheet = () => {
+      // Set state on close so when it opens again this is what you see
+      setState("select");
+      bottomSheetRef.current?.close();
+    };
+
+    const handlePressImport = async () => {
+      setState("importing");
+      let results;
+      try {
+        results = await DocumentPicker.getDocumentAsync();
+      } catch (err) {
+        setState("error");
+        return;
+      }
+
+      if (results.type === "cancel") {
+        closeSheet();
+        return;
+      }
+
+      try {
+        const {
+          import: { id: importId },
+          style,
+        } = await api.maps.importTileset(results.uri);
+
+        if (style) {
+          addMapImport({ styleId: style.id, importId });
+        }
+
+        closeSheet();
+      } catch (err) {
+        console.log("IMPORT ERROR", err);
+        const parsedError = await extractHttpErrorResponse(err)?.json();
+
+        if (
+          parsedError &&
+          parsedError.statusCode >= 400 &&
+          parsedError.statusCode < 500
+        ) {
+          setState("error");
+          return;
+        }
+
+        // TODO: How to handle different kind of error?
+        closeSheet();
+      }
+    };
+
+    const BackdropComponent =
+      state === "importing" ? BackdropNonPressable : BackdropPressable;
+
+    const bottomSheetContent =
+      state === "select" ? (
+        <SelectImportContent
+          onPressClose={closeSheet}
+          onPressImport={handlePressImport}
+        />
+      ) : state === "importing" ? (
+        <ImportingContent />
+      ) : (
+        <FileErrorContent onPressClose={closeSheet} />
+      );
+
+    return (
+      <BottomSheet
+        index={-1}
+        enableContentPanningGesture={false}
+        enableHandlePanningGesture={false}
+        handleComponent={() => null}
+        ref={bottomSheetRef}
+        backdropComponent={BackdropComponent}
+        snapPoints={snapPoints}
+      >
+        <BottomSheetView
+          onLayout={updateSheetHeight}
+          style={styles.bottomSheetView}
+        >
+          {bottomSheetContent}
+        </BottomSheetView>
+      </BottomSheet>
+    );
+  }
+);
+
+export default MapImportBottomSheet;
+
+function useSnapPointsCalculator() {
```

Comment:
Should we just use the `useSnapPointsCalculator` from `BottomSheetModal` in shared components?